test/integration/targets/lightsail/tasks/main.yml

```diff
@@ -0,0 +1,136 @@
+---
+
+- name: Integration test for lightsail module
+  block:
+
+    # ==== Setup ==================================================
+
+    - name: Set connection information for all tasks
+      set_fact:
+        aws_connection_info: &aws_connection_info
+          aws_access_key: "{{ aws_access_key }}"
+          aws_secret_key: "{{ aws_secret_key }}"
+          security_token: "{{ security_token }}"
+          region: "{{ aws_region }}"
+      no_log: yes
+
+    # ==== Tests ===================================================
+
+    - name: Create a new instance
+      lightsail:
+        name: "{{ instance_name }}"
+        zone: "{{ zone }}"
+        blueprint_id: amazon_linux
+        bundle_id: nano_2_0
+        key_pair_name: "{{ keypair_name }}"
+        <<: *aws_connection_info
+      register: result
+
+    - assert:
+        that:
+          - result.changed == True
+          - "'instance' in result and result.instance.name == instance_name"
+
+    - name: Make sure create is idempotent
+      lightsail:
+        name: "{{ instance_name }}"
+        zone: "{{ zone }}"
+        blueprint_id: amazon_linux
+        bundle_id: nano_2_0
+        key_pair_name: "{{ keypair_name }}"
+        <<: *aws_connection_info
+      register: result
+
+    - assert:
+        that:
+          - result.changed == False
+
+    - name: Start the running instance
+      lightsail:
+        name: "{{ instance_name }}"
+        state: running
+        <<: *aws_connection_info
+      register: result
+
+    - assert:
+        that:
+          - result.changed == False
```

Comment:
It would be nice to check that the state is running or pending here as well (like you do in the next start task further down).  You could even add a `wait: yes` to the previous task to validate that the waiter function does the right thing, and then only assert on `running`.

Reply:
okay, I will add the `wait: yes`. 👍

Reply:
I don't think the `wait` option is working as one would expect. The `wait` option defaults to `yes` for this module. I ran a task to create instance with `wait: yes`, but the module returned with `state=pending`.
Looking at the [code](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/lightsail.py#L316), wait is being used to let the instance come out of a transient state (pending, stopping).  The `create_instance` function is not looking at `wait` at all.

Reply:
So should I add a failing test or should I add it and comment it out for now?

Reply:
The description for the `wait` option in the module documentation is also incorrect. It says -

> Wait for the instance to be in state 'running' before returning. If wait is "no" an ip_address may not be returned

But that is not the case. I ran a task to create instance and it returned this as the state -
`        "state": {
            "code": 0,
            "name": "pending"
        },
`

Reply:
Ah apologies, I naively trusted the module docs.  For now let's just assert that the instance is in ['pending', 'running'].  I've opened #63869 about the lack of wait support for create actions.

Reply:
I have added the test case to check the instance state is in ['pending', 'running'].